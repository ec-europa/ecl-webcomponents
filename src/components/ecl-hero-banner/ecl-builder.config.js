const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-hero-banner-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-hero-banner-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-hero-banner-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-hero-banner-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
      },
    },
  ],
};
