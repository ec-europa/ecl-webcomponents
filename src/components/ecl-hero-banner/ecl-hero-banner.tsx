import { Component, Prop, h } from '@stencil/core';

@Component({
  tag: 'ecl-hero-banner',
  styleUrls: {
    ec: './build/styles/ecl-hero-banner-ec.css',
    eu: './build/styles/ecl-hero-banner-eu.css',
  },
  shadow: false,
  scoped: true,
})
export class EclHeroBanner {
  @Prop() theme: string = 'ec';
  @Prop() styleClass: string;
  @Prop() external: boolean = false;
  @Prop() variant: string = 'display';
  @Prop() fullWidth: boolean = false;
  @Prop() bannerTitle: string;
  @Prop() image: string;
  @Prop() ctaLabel: string;
  @Prop() ctaLink: string;
  @Prop() centered: boolean = true;
  @Prop() credit: string;

  getClass(): string {
    const styleClasses = [
      `ecl-hero-banner`,
      `ecl-hero-banner--${this.variant}`,
      this.styleClass
    ];

    if (this.centered) {
      styleClasses.push('ecl-hero-banner--centered');
    }
    if (this.fullWidth) {
      styleClasses.push('ecl-hero-banner--full-width');
    }

    return styleClasses.join(' ');
  }

  render() {
    const imageBanners = [
      'image',
      'image-gradient',
      'image-shade',
    ];
    return (
      <section class={this.getClass()}>
      { imageBanners.includes(this.variant) && this.image ?
        <div
          class="ecl-hero-banner__image"
          style={{ backgroundImage: "url(" + this.image + ")"}}
        >
        </div> : '' 
      }
      { imageBanners.includes(this.variant) && this.credit ?
        <div class="ecl-hero-banner__credit">{this.credit}</div> : ''
      }
        <div class="ecl-container">
          <div class="ecl-hero-banner__container">
            <div class="ecl-hero-banner__content">
            { this.bannerTitle ?
              <div class="ecl-hero-banner__title">{this.bannerTitle}</div> : ''
            }
              <p class="ecl-hero-banner__description">
                <span class="ecl-hero-banner__description-text">
                  <slot></slot>
                </span>
              </p>
            { this.ctaLabel && this.ctaLink ?
              <div class="ecl-hero-banner__cta">
                <ecl-link
                  variant="cta"
                  path={this.ctaLink}
                  style-class={`ecl-hero-banner__link-cta sc-ecl-hero-banner-${this.theme}`}
                >
                  {this.ctaLabel}
                  <ecl-icon
                    icon="corner-arrow"
                    slot="icon-after"
                    transform="rotate-90"
                  ></ecl-icon>
                </ecl-link>
              </div>
            : '' }
            </div>
          </div>
        </div>
      </section>
    );
  }
}
