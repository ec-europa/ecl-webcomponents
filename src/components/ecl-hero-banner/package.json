{
  "name": "@ecl/web-components-hero-banner",
  "author": "European Commission",
  "license": "EUPL-1.2",
  "version": "0.1.0-alpha.0",
  "description": "ECL Page Banner",
  "main": "ecl-hero-banner.tsx",
  "private": true,
  "scripts": {
    "build:styles": "cross-env NODE_ENV=production ecl-builder styles",
    "build": "run-s clean build:*",
    "clean": "rimraf build",
    "prepublish": "npm run build"
  },
  "devDependencies": {
    "cross-env": "7.0.3",
    "@ecl/builder": "3.7.1",
    "@ecl/vanilla-component-hero-banner": "3.7.1"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/planctus/ecl-webcomponents.git"
  },
  "bugs": {
    "url": "https://github.com/planctus/ecl-webcomponents/issues"
  },
  "homepage": "https://github.com/planctus/ecl-webcomponents",
  "keywords": [
    "ecl",
    "europa-component-library",
    "design-system",
    "web-components"
  ]
}
