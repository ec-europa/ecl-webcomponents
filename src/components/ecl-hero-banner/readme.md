# ecl-hero-banner



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description | Type      | Default     |
| ------------- | -------------- | ----------- | --------- | ----------- |
| `bannerTitle` | `banner-title` |             | `string`  | `undefined` |
| `centered`    | `centered`     |             | `boolean` | `true`      |
| `ctaLabel`    | `cta-label`    |             | `string`  | `undefined` |
| `ctaLink`     | `cta-link`     |             | `string`  | `undefined` |
| `external`    | `external`     |             | `boolean` | `false`     |
| `fullWidth`   | `full-width`   |             | `boolean` | `false`     |
| `image`       | `image`        |             | `string`  | `undefined` |
| `styleClass`  | `style-class`  |             | `string`  | `undefined` |
| `theme`       | `theme`        |             | `string`  | `'ec'`      |
| `variant`     | `variant`      |             | `string`  | `'display'` |


## Dependencies

### Depends on

- [ecl-link](../ecl-link)
- [ecl-icon](../ecl-icon)

### Graph
```mermaid
graph TD;
  ecl-hero-banner --> ecl-link
  ecl-hero-banner --> ecl-icon
  ecl-link --> ecl-icon
  style ecl-hero-banner fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
