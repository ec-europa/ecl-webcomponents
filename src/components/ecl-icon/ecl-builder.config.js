const path = require('path');
const pkg = require('./package.json');

const nodeModules = path.resolve(__dirname, '../../../node_modules');
const outputFolder = './build';

// SCSS includePaths
const includePaths = [nodeModules];

const banner = `${pkg.name} - ${pkg.version} Built on ${new Date().toISOString()}`;

module.exports = {
  copy: [
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-social-media-icons/dist/sprites/icons-social-media.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-ec-icons/dist/sprites/icons.svg')],
      to: path.resolve(outputFolder, 'images/ec'),
      options: { up: true },
    },
    {
      from: [path.resolve(nodeModules, '@ecl/resources-eu-icons/dist/sprites/icon.svg')],
      to: path.resolve(outputFolder, 'images/eu'),
      options: { up: true },
    },
  ],
  styles: [
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-ec.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-ec.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
    {
      entry: path.resolve(__dirname, 'src/styles/ecl-icon-eu.scss'),
      dest: path.resolve(outputFolder, 'styles/ecl-icon-eu.css'),
      options: {
        banner,
        includePaths,
        sourceMap: false,
        minify: true,
      },
    },
  ],
};
