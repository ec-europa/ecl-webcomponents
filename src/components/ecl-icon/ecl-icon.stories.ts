import iconPathEc from '@ecl/resources-ec-icons/dist/sprites/icons.svg';
import iconPathEu from '@ecl/resources-eu-icons/dist/sprites/icons.svg';
import iconsAllEc from '@ecl/resources-ec-icons/dist/lists/all.json';

export default {
  title: 'Components/icon',
};

const Template = (args) => `<ecl-icon 
                              icon="${args.icon}"
                              color="${args.color}"
                              size="${args.size}"
                              transform="${args.transform}"
                              path="${args.theme === 'ec' ? iconPathEc : iconPathEu}"
                            ></ecl-icon>`;

export const Icon = Template.bind({});
Icon.storyName = 'default';
Icon.args = {
  theme: 'ec',
  icon: 'close',
  transform: '',
  color: 'default',
  size: '2xl',
};
Icon.argTypes = {
  theme: {
    type: { name: 'select'},
    options: ['ec', 'eu'],
  },
  icon: {
    type: { name: 'select' },
    options: iconsAllEc,
    description: "Add an icon",
  },
  size: {
    type: { name: 'select' },
    options: ['2xs','xs', 's', 'm', 'l', 'xl', '2xl'],
    description: "icon size",
  },
  color: {
    type: { name: 'select' },
    options: ['default', 'primary', 'inverted'],
    description: "icon color",
  },
  transform: {
    type: { name: 'select' },
    description: 'Button icon transform',
    options: [
      'rotate-90',
      'rotate-180',
      'rotate-270',
      'flip-horizontal',
      'flip-vertical',
    ],
  },
};