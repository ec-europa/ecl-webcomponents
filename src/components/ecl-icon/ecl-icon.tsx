import { Component, Prop, h } from '@stencil/core';
import getAssetPath from "../../utils/assetPath";

@Component({
  tag: 'ecl-icon',
  styleUrls: {
    ec: './build/styles/ecl-icon-ec.css',
    eu: './build/styles/ecl-icon-eu.css',
  },
  shadow: false,
  scoped: true,
  assetsDirs: ['build'],
})

export class EclIcon {
  @Prop() styleClass: string = '';
  @Prop() icon: string = '';
  @Prop() size: string = 'xs';
  @Prop() color: string = '';
  @Prop() titleTag: string = '';
  @Prop({ mutable: true }) path: string;
  @Prop() transform: string = '';
  @Prop({ mutable: true }) theme: string;
  @Prop() sprite: string = '';

  componentWillLoad() {
    this.theme = document.documentElement.getAttribute('data-ecl-theme') ?? (this.theme || 'ec');
    
    if (!this.sprite) {
      this.path = getAssetPath(`./build/images/${this.theme}/icons.svg`);
    } else {
      this.path = getAssetPath(`./build/images/${this.theme}/${this.sprite}.svg`);
    }
  }

  getClass(): string {
    const styleClasses = [
      `ecl-icon`,
      `ecl-icon--${this.size}`,
      this.styleClass,
    ];
    if (this.color) {
      styleClasses.push(
        `ecl-icon--${this.color}`
      );
    }
    if (this.transform) {
      styleClasses.push(
        `ecl-icon--${this.transform}`
      );
    }

    return styleClasses.join(' ');
  }

  render() {
    return (
      <svg class={this.getClass()}>
      { this.titleTag ? <title>{this.titleTag}</title> : '' } 
        <use xlinkHref={`${this.path}#${this.icon}`}></use>
      </svg>
    )
  }
}