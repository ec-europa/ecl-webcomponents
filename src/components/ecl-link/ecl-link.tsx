import { Component, h, Prop, Element} from '@stencil/core';

@Component({
  tag: 'ecl-link',
  styleUrls: {
    ec: './build/styles/ecl-link-ec.css',
    eu: './build/styles/ecl-link-eu.css',
  },
  shadow: false,
  scoped: false,
  assetsDirs: ['build'],
})

export class EclLink {
  @Element() el: HTMLElement;
  @Prop() theme: string = 'ec';
  @Prop() path: string;
  @Prop() styleClass: string;
  @Prop() variant: string = 'default';
  @Prop() titleAttr: string;
  @Prop() external: boolean = false;

  getClass(): string {
    const styleClasses = [
      `ecl-link`,
      `ecl-link--${this.variant}`,
      this.styleClass
    ];

    if (this.el.querySelector('ecl-icon')) {
      styleClasses.push('ecl-link--icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        styleClasses.push('ecl-link--icon-after');
      } else {
        styleClasses.push('ecl-link--icon-before');
      }
    }

    return styleClasses.join(' ');
  }

  componentDidRender() {
    if (this.el.querySelector('ecl-icon')) {
      this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon');
      if (this.el.querySelector('ecl-icon').getAttribute('slot') === 'icon-after') {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--after');
      } else {
        this.el.querySelector('ecl-icon svg').classList.add('ecl-link__icon--before');
      }
    }
  }

  getExternal() {
    return (
      <ecl-icon 
        icon="external"
        style-class={`ecl-link__icon ecl-link__icon--after sc-ecl-link-${this.theme}`}
        size = "2xs"
      >
      </ecl-icon>
    )  
  }

  render() {
    return (
      <a
        class={this.getClass()}
        href={this.path}
        title={this.titleAttr}
      >
        <slot name="icon-before"></slot>
        <slot></slot>
        <slot name="icon-after">{ this.external ? this.getExternal() : '' }</slot>
      </a>
    );
  }
}
