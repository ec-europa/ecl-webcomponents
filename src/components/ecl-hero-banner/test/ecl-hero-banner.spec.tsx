import { newSpecPage } from '@stencil/core/testing';
import { EclHeroBanner } from '../ecl-hero-banner';
import { EclLink } from '../../ecl-link/ecl-link';
import { EclIcon } from '../../ecl-icon/ecl-icon';

describe('ecl-page-banner', () => {
  it('renders', async () => {
    const page = await newSpecPage({
      components: [EclHeroBanner, EclLink, EclIcon],
      html: `<ecl-hero-banner
              variant="primary"
              theme="ec"
              cta-link="/example.html"
              cta-label="Subscribe"
              image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg"
              banner-title="EU Budget for the future"
            >
              Innovation, economy, environment and geopolitics
            </ecl-hero-banner>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});

