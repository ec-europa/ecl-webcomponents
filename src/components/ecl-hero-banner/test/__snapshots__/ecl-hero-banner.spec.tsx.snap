// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecl-hero-banner renders 1`] = `
<ecl-hero-banner banner-title="EU Budget for the future" credit="Â© Copyright or credit" cta-label="Subscribe" cta-link="/example" image="https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg" theme="ec" variant="primary">
  <!---->
  <section class="ecl-hero-banner ecl-hero-banner--centered ecl-hero-banner--primary">
    <div class="ecl-container">
      <div class="ecl-hero-banner__container">
        <div class="ecl-hero-banner__content">
          <div class="ecl-hero-banner__title">
            EU Budget for the future
          </div>
          <p class="ecl-hero-banner__description">
            <span class="ecl-hero-banner__description-text">
              Innovation, economy, environment and geopolitics
            </span>
          </p>
          <div class="ecl-hero-banner__cta">
            <ecl-link style-class="ecl-hero-banner__link-cta sc-ecl-hero-banner-ec">
              <!---->
              <a class="ecl-hero-banner__link-cta ecl-link ecl-link--cta ecl-link--icon ecl-link--icon-after sc-ecl-hero-banner-ec" href="/example">
                <ecl-icon slot="icon-after">
                  <svg class="ecl-icon ecl-icon--rotate-90 ecl-icon--xs ecl-link__icon ecl-link__icon--after">
                    <use xlink:href="/build/images/ec/icons.svg#corner-arrow"></use>
                  </svg>
                </ecl-icon>
                Subscribe
                <slot-fb hidden="" name="icon-after"></slot-fb>
              </a>
            </ecl-link>
          </div>
        </div>
      </div>
    </div>
  </section>
</ecl-hero-banner>
`;
